### ISSUE TITLE
value of attribute target, ref, and ed producing errors with 3.1.0

### ISSUE THREAD

**Issue by jeffreycwitt** [TEIC/TEI#1601]
Today I updated my odd from 2.8.0 to 3.1.0 and now I'm getting the following errors in oxygen.

value of attribute "target" is invalid 
value of attribute "ref" is invalid
value of attribute "ed" is invalid

Examples uses of target, ref, and ed look like: 

`<ref target="http://creativecommons.org/licenses/by-nc-nd/3.0/">Creative Commons Attribution-NonCommercial-NoDerivs 3.0 License</ref>`

`<name ref="#Wodeham">Adam</name>`

`<lb ed="#E"/>`

If I change the schema spec back to 2.8.0 all the errors go away. 

The current odd is here: https://github.com/lombardpress/lombardpress-schema/blob/develop/src/diplomatic.xml

Any ideas where I'm going wrong

**Comment by lb42** [comment]
That file isn't valid TEI XML. It's not even well formed! Try removing 
the stray "-->" at line 154 for a start...

**Comment by jeffreycwitt** [comment]
Ok, well I've removed the "-->" on line 154

Oxygen tells me it's now well formed and TEI valid, and I'm having the same problem. 

I'm not suggesting this is a problem with the TEI guidelines, I'm just reporting my problem and looking for guidance. If this is not the right venue for help, let me know and I can ask my question elsewhere. 

The point, however, is simply that if I use a 2.8 declaration in the schemaSpec, I don't have a problem, and if I change it to 3.1 I do have problem. I find this strange, and I'm just trying to figure out what I'm doing wrong.

Any help is much appreciated.

**Comment by martindholmes** [comment]
I created a RelaxNG schema from your ODD in Oxygen 18.1 and it seems to be fine, but without a sample document to validate against it, I can't be sure. Can you point me at the document that has these problem items in them?

In the ODD, I did notice this:
~~~~~
<change who="#JW" when="2012-05-08" n="0.0.0-dev"></change>
 ~~~~~

There's no element in the file with @xml:id="JW", which is an obvious error; could what you're seeing in the validation of the other documents be similar instances?

**Comment by lb42** [comment]
Apologies if my hasty reply last night was not very helpful. In fact this is quite a complicated bug and, as you infer, it's down to a change between 2.8 and 3.1, or 2.8 and the present. The change is that we are now using datatypes called teidata.xxx rather than data.xxx : your ODD refers to two datatypes by the old names, and this is the fundamental issue. It *ought* not to matter since the old definitions are still hanging around, but I suspect that some optimisation in the ODD processing stylesheets is ignoring them. We probably haven't done enough to alert ODD writers to the need to watch out for this problem, though it is mentioned in my wonderful tutorial cited below.

Fortunately in your case, you can simply delete the datatype references. The first one, on your changes to note/@type, is redundant since you're not changing the datatype. The second one, in your added attDef for note/@hand is also redundant (and results in an erroneous schema with two hand attributes) since this attribute is already defined for hand: just delete the whole attDef.

With those changes, I generated a valid RNC schema, with no dangling datatype references. As Martin says, it's hard to validate this further without a test data file.

ps: you might find my tutorial at http://teic.github.io/TCW/purifyDoc.html interesting if you're updating your ODD

**Comment by jeffreycwitt** [comment]
@lb42 and @martindholmes many thanks for your helpful responses and clarifications. The purifyDoc tutorial is particularly helpful for understanding the new changes that have come with 3.0.0

In the end, I think the Oxygen framework I had installed was incompatible with my attempts to use 3.1 of the guidelines. 

I followed James' tutorial http://blogs.it.ox.ac.uk/jamesc/2014/04/02/auto-update-your-tei-framework-in-oxygen/ to auto update my framework, and 'voila' the errors retreated. 

Sorry to trouble you with my question, but I very much appreciate the help. :)
